examples/QuickJournal/QuickJournal/ViewModels/JournalEntriesViewModel.cs

```diff
@@ -1,66 +1,77 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Threading.Tasks;
 using System.Windows.Input;
+using QuickJournal.Models;
+using QuickJournal.Services;
 using Realms;
+using Xamarin.CommunityToolkit.ObjectModel;
 using Xamarin.Forms;
 
-namespace QuickJournal
+namespace QuickJournal.ViewModels
 {
-    public class JournalEntriesViewModel
+    public class JournalEntriesViewModel : BaseViewModel
     {
-        // TODO: add UI for changing that.
-        private const string AuthorName = "Me";
-
-        private Realm _realm;
+        readonly JournalEntry selectedEntry = null;
+        readonly Realm realm;
 
         public IEnumerable<JournalEntry> Entries { get; private set; }
 
         public ICommand AddEntryCommand { get; private set; }
-
+        public ICommand EditEntryCommand { get; private set; }
         public ICommand DeleteEntryCommand { get; private set; }
 
         public INavigation Navigation { get; set; }
 
+        public JournalEntry SelectedEntry
+        {
+            get => selectedEntry;
+            set
+            {
+                EditEntryCommand.Execute(value);
+                OnPropertyChanged();
+            }
+        }
+
         public JournalEntriesViewModel()
         {
-            _realm = Realm.GetInstance();
+            realm = Realm.GetInstance();
+            Entries = realm.All<JournalEntry>();
 
-            Entries = _realm.All<JournalEntry>();
+            InitCommands();
+        }
 
-            AddEntryCommand = new Command(AddEntry);
+        private void InitCommands()
```

Comment:
[nit] I'm not sure how I feel about this `InitCommands` method. Generally, I'd prefer stuff that only ever gets called once to be in the ctor rather than having a dedicated method for it. While I understand the sentiment of breaking up a large method into smaller components, it does feel a  little clunky and forces us to have a setter for every command. If we folded it into the ctor, we could have a getter-only property instead.

Reply:
I agree, probably for such a simple app all of this small methods are unnecessary